sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/KalixServer.scala

```diff
@@ -371,6 +386,26 @@ case class KalixServer(applicationContext: ApplicationContext, config: Config) {
         kalixBeanFactory.getBean(clz)
       })
 
+  private def workflowProvider[S, E <: Workflow[S]](clz: Class[E]): WorkflowProvider[S, E] =
+    ReflectiveWorkflowProvider.of(
+      clz,
+      messageCodec,
+      context => {
+        if (hasContextConstructor(clz, classOf[WorkflowContext])) {
+          WorkflowContextFactoryBean.set(context)
+        }
+
+        val webClientProviderHolder = WebClientProviderHolder(context.materializer().system)
+
+        if (hasContextConstructor(clz, classOf[KalixClient])) {
+          kalixClient.setWebClient(webClientProviderHolder.webClientProvider.localWebClient)
+          // we only have one KalixClient, but we only set it to the ThreadLocalFactoryBean
+          // when building actions, because it's only allowed to inject it in Actions
```

Comment:
time to update this commend. We only allowed it to be inject in Actions and Workflows

Reply:
updated